WooCommerce/Classes/ViewRelated/BottomButtonContainer/BottomButtonContainerView.swift

```diff
@@ -0,0 +1,64 @@
+import UIKit
+
+/// Contains a button with insets to be displayed at the bottom of a view.
+///
+final class BottomButtonContainerView: UIView {
+    struct ViewModel {
+        /// Allows the view model to configure and style the button.
+        let configureButton: (UIButton) -> Void
+
+        /// Called when the button is tapped.
+        let onButtonTapped: (UIButton) -> Void
+    }
+
+    private let button: UIButton = UIButton(type: .custom)
+
+    private let viewModel: ViewModel
+
+    init(viewModel: ViewModel) {
+        self.viewModel = viewModel
+        super.init(frame: .zero)
+
+        translatesAutoresizingMaskIntoConstraints = false
+        configureContainerView()
+        configureButton()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+private extension BottomButtonContainerView {
+    func configureContainerView() {
+        backgroundColor = .basicBackground
+
+        let topBorderView = UIView.createBorderView()
+        addSubview(topBorderView)
+        NSLayoutConstraint.activate([
+            topBorderView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            topBorderView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            topBorderView.topAnchor.constraint(equalTo: topAnchor)
+        ])
+    }
```

Comment:
From what I understand, this is the only unique thing that we're getting from `BottomButtonContainerView`. And if we want to reuse this in #2204, we'll have to change #2204 to have a `configureButton` callback too, right? 

Also, with the `configureButton`, if we want something like the Products button that's implemented here, we'll have to _copy_ this part too, right?

https://github.com/woocommerce/woocommerce-ios/blob/46ddafb736872b5f90f9a29a42012a87f9c9f763/WooCommerce/Classes/ViewRelated/Products/Edit%20Product/ProductFormViewController.swift#L164-L172

I wonder if we can make this a more closed **standard** component. I'm guessing that we are probably not gonna have a lot of button styles. So, instead of exposing a `configureButton`, could we perhaps expose a `Style`? 

```swift
enum ButtonStyle {
    // For #2204
    case primary
    // For #2210
    case linkButton
}
```

And then the styling would just be handled **inside** this class. Consumers would only need to:

```swift
let containerView = BottomButtonContainerView(
    style: .primary,
    title: "The Title",
    image: .theImage,
    onButtonTapped: { ... }
)
```

This still allows us to have a _reusable_ component but standardized for Woo styling. 

What do you think?

Reply:
> From what I understand, this is the only unique thing that we're getting from BottomButtonContainerView.

yes, the current constraint setup is specific for usage at the bottom of a view. if we have similar design but at the top for example, we could add a config in the view model and set up constraints accordingly

Reply:
> I wonder if we can make this a more closed standard component. I'm guessing that we are probably not gonna have a lot of button styles. So, instead of exposing a configureButton, could we perhaps expose a Style?

excellent idea! I updated with the addition of button style in https://github.com/woocommerce/woocommerce-ios/pull/2210/commits/25aba07d6205ab492ee6660072ee3830032e720c 😄